Perf testing and tco integration

VanshChitransh opened the pull request:
# Performance Testing and TCO Estimation Implementation

This PR implements comprehensive performance testing and TCO (Total Cost of Ownership) estimation tools for Mojafos deployment as per the project requirements.

## Features Implemented:

1. **Performance Testing with JMeter**
   - Automated script for PaymentHub EE testing
   - Configurable user count, duration, and other parameters
   - HTML report generation

2. **vNext Performance Tools Integration**
   - Integration with Mojaloop vNext performance tools from ndogo-loop
   - Fallback to JMeter-based testing when native tools are unavailable
   - Consolidated reporting

3. **TCO Estimation**
   - Cloud infrastructure cost estimation based on performance metrics
   - Support for various instance types and regions
   - Comprehensive cost breakdown (compute, storage, network)
   - Long-term cost projections

4. **Master Orchestration**
   - Combined execution of all testing components
   - Unified reporting
   - Configurable test scenarios

## Screenshots:

<img width="1089" alt="image" src="https://github.com/user-attachments/assets/c42c1fc5-bdd5-47d9-a457-5f0c8b54250b" />
<img width="427" alt="image" src="https://github.com/user-attachments/assets/34cc38a1-d815-4b80-9b22-1e2120a98c75" />
<img width="631" alt="image" src="https://github.com/user-attachments/assets/e354106c-7e91-4f26-836a-346014875c02" />


## Testing:

The implementation has been tested with sample data, and all components are functioning correctly. The scripts are compatible with standard Bash environments and require JMeter, jq, and bc for full functionality.

## Documentation:

Detailed documentation has been added to the README.md file in the performance-testing directory, covering all aspects of the implementation, usage instructions, and troubleshooting tips.

Commit:
Add performance testing and TCO estimation tools for Mojafos deployment

Commit:
Update performance testing scripts for compatibility and add sample data testing

DavidH-1 commented:
@tdaly61 please also review

DavidH-1 commented:
@VanshChitransh why does this introduce references to Mojafos?  The product is Mifos Gazelle.  Also do the tests include the MIfosX instance or just VNext and PH-EE

VanshChitransh commented:
Thank you for your feedback @DavidH-1!

1. Regarding the "Mojafos" terminology - you're absolutely right. I incorrectly used "Mojafos" instead of "Mifos Gazelle" throughout the implementation. I'll update all references to use the correct product name.

2. Regarding testing coverage - the current implementation focuses on PaymentHub EE and vNext integration. MifosX testing is not explicitly included yet. Would you like me to extend the implementation to include MifosX testing as well? If so, I'd be happy to add that in a follow-up PR or update this one.

Abhinavcode13 commented:
Hi @VanshChitransh, have you had a chance to deploy the ndogo-loop for the vNext beta? Check [here](https://github.com/tdaly61/ndogo-loop/tree/dev)

VanshChitransh commented:
@Abhinavcode13 Thanks for your question! 

I haven't deployed ndogo-loop for vNext beta testing yet. The current implementation includes integration code that will connect with ndogo-loop's performance tools, but it also has fallback mechanisms when direct integration isn't available.

Would you like me to test with an actual ndogo-loop deployment as part of this PR? If so, I'd be happy to do that and provide evidence of the integration working end-to-end.

I'm also working on fixing the terminology issues (changing "Mojafos" to "Mifos Gazelle") throughout the codebase. I'll push those changes shortly.

VanshChitransh commented:
Accidentally requested review — this PR is still a work in progress. Sorry for the noise!

Commit:
Fix: Replace Mojafos with correct product name Mifos Gazelle

VanshChitransh commented:
@Abhinavcode13 and @DavidH-1, thank you for your review and feedback!

I've addressed the following items:

1. **Terminology Fix**: I've pushed commit 90b6e71 that replaces all instances of "Mojafos" with the correct product name "Mifos Gazelle" throughout the codebase.

2. **ndogo-loop Integration**: The current implementation includes the integration framework for ndogo-loop's performance tools with appropriate fallback mechanisms. The integration works by:
   - Cloning the ndogo-loop repository
   - Checking for available performance testing scripts
   - Using those scripts if available or falling back to JMeter tests

3. **Testing Scope**: As mentioned earlier, the current implementation focuses on PaymentHub EE and vNext integration. MifosX testing is not included in the current scope.

Before this PR can be considered ready for final review, I'd like to clarify:

1. Should I deploy and test with ndogo-loop before finalizing this PR, or is the integration framework sufficient for now?
2. Would you like me to extend the implementation to include MifosX testing in this PR, or should that be addressed in a follow-up PR?
3. Are there any other changes or improvements you'd like to see before considering this PR for merging?

I'll keep this PR in draft status until we've addressed these questions.
